tc/sgx/trusted_worker_manager/enclave/work_order_data_handler.cpp

```diff
@@ -131,9 +131,18 @@ namespace tcf {
 
     }
 
-    void WorkOrderDataHandler::VerifyInputHash(ByteArray input_data,
-		                               ByteArray input_hash) {
-        // Do nothing at the phase 1
+    tcf_err_t WorkOrderDataHandler::VerifyInputHash(ByteArray input_data,
+                                                    ByteArray input_hash) {
+        tcf_err_t verify_status = TCF_SUCCESS;
+        ByteArray hash = tcf::crypto::ComputeMessageHash(input_data);
+        if (std::equal(hash.begin(), hash.end(), input_hash.begin())) {
```

Comment:
Question: `input_hash` is coming from the external source. `hash` is internally calculated, will the equal method take care if the `input_hash` is of smaller length than `hash`?

Reply:
Yes, std::equal function takes care if input_hash is of smaller length